### PR TITLE
Set date format

### PR DESCRIPTION
**Pull request by bianca-actu**
- Upgrade to poetry 1.0
- Add test for setting log format
- Allow setting the date format
- Remove assumption in Log.__getattr__ that it's not a top level logger name

Passes tests and fmt checks with Python 3.8.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -28,4 +28,4 @@ flake8 = "*"
 twine = "*"
 wheel = "*"
 isort = "*"
-black = {python=">3.6", version=">=18.9b0", allow_prereleases=true}
+black = {python=">3.6", version="^18.9b0", allow-prereleases=true}
```
Comment (djrobstep) [inline pyproject.toml]: These days poetry seems to install `black` fine without the `allow-pre` parameter at all (I guess it derives it from the version spec?)

File: logx/gumph.py [hunk 2]
```diff
@@ -132,6 +135,11 @@ def __getattr__(self, method, *args, **kwargs):
         try:
             self.get_handler(logger_name)
         except KeyError:
-            self.add_std_handler(logger_name)
+            try:
```
Comment (djrobstep) [inline logx/gumph.py]: Thinking maybe this logic should be handled within `get_handler` (via adding an option to get both top-level and non-top-level handlers). Seems a bit cleaner than nested try/catches. What do you reckon?

Comment (bianca-actu) [reply logx/gumph.py]: Yeah, thanks - I agree. I wasn't particularly happy with it to start with.

Comment (bianca-actu) [reply logx/gumph.py]: Wait, no - the reason I did that was so I could get the right the `logger_name` without having to change the public interface. I'll make another function that returns a tuple of the handler and logger name instead.

File: pyproject.toml (hunk 1 above)
Comment (bianca-actu) [reply pyproject.toml]: OK, I'm happy to change this if you'd like me to?

File: logx/gumph.py (hunk 2 above)
Comment (bianca-actu) [reply logx/gumph.py]: Check out the new changes.

File: pyproject.toml (hunk 1 above)
Comment (djrobstep) [reply pyproject.toml]: Yeah cool, might as well remove it

File: logx/gumph.py (hunk 2 above)
Comment (djrobstep) [reply logx/gumph.py]: Looks good!

**Comment by djrobstep** [comment]
Great stuff, merging!

**Comment by bianca-actu** [comment]
@djrobstep Thanks, it looks like the CI failed, changing the poetry install stage to this should do the trick:
`sudo pip3 install poetry==1.0`
I should have been clearer that the pyproject.toml the new poetry version is not compatible.

**Comment by bianca-actu** [comment]
@djrobstep We'd like to use this feature soon (I'm currently modifying our applications to use it) so would really appreciate if you could publish a release in the near future.

**Comment by djrobstep** [comment]
I fixed the build so the latest release should have this functionality in it... If I have missed something let me know

**Comment by bianca-actu** [comment]
Cool, thanks

**Pull request merged**